whitepapers/0010_ibc_generic_messaging

```diff
@@ -0,0 +1,92 @@
+- Define a custom IBC specification for passing wormhole messages between IBC chains and wormchain.
+- Ensure this design is backwards-compatible with existing Cosmos integrations.
+- Ensure this design does not violate any of Wormhole's existing security assumptions.
+
+## Non-Goals
+
+This document does not propose new cosmos networks for Wormhole to support. It is focused on the technical design of using IBC generic messaging to reduce the operational load on wormhole guardians.
+
+This document is also not meant to describe how wormhole can be scaled beyond the cosmos ecosystem.
+
+## Overview
+
+Currently wormhole guardians run full nodes for every chain that wormhole is connected to. This is done for maximum security and decentralization. Since each guardians runs a full node for each chain, they are able to independently verify the authenticity of wormhole messages that are posted on different blockchains. However, running full nodes has its drawbacks. Specifically, adding new chains to wormhole has a high operational cost per chain, which makes it difficult to scale wormhole.
+
+Luckily, we can leverage standards such as IBC to scale wormhole's support for the cosmos ecosystem. Since Cosmos IBC messages are trustlessly verified by tendermint light clients, we can pass wormhole messages from any cosmos chain over IBC to wormchain, which will then emit that message for the wormhole guardians to pick up. This way, the wormhole guardians only need to run a full node for wormchain to be able to verify the authenticity of messages on all other IBC-enabled cosmos chains.
+
+## Detailed Design
+
+### External Chain -> Cosmos Chain
+
+This will work exactly the same way it works today. We will deploy our wormhole cosmwasm contract stack to every cosmos chain we want to support. Wormhole relayers will be able to post VAAs produced for any source chain directly to the cosmos destination chain.
+
+### Cosmos Chain -> External Chain
+
+Typically, the wormhole core bridge contract emits a message which the guardians then pick up from their full nodes.
+
+For cosmos chains, we update the core bridge contract to instead send this message over IBC to wormchain. Then a wormchain contract receives the message to emit and actually emits it, which the guardians then pick up.
+
+Specifically, we implement two new cosmwasm smart contracts: `wormhole-ibc` and `wormchain-ibc-receiver`.
+
+The `wormhole-ibc` contract is meant to replace the `wormhole` core bridge contract on cosmos chains. It imports the `wormhole` contract as a library and delegates core functionality to it before and after running custom logic:
+- During execution, it delegates all logic to the core bridge library. After the core bridge library has finished execution, it will send the response as an IBC message to the `wormchain-ibc-receiver` contract <b>if and only if</b> the execution message is of type `wormhole::msg::ExecuteMsg::PostMessage`.
```

Comment:
```suggestion
- During execution, it delegates all logic to the core bridge library. After the core bridge library has finished execution, it will send the response as an IBC message to the `wormchain-ibc-receiver` contract **if and only if** the execution message is of type `wormhole::msg::ExecuteMsg::PostMessage`.
```

You can get **bold** text in markdown using `**bold text here**` instead of inline html.